internal/agent/service/collector.go

```diff
@@ -29,15 +29,17 @@ import (
 )
 
 type Collector struct {
-	dataDir        string
-	credentialsDir string
-	once           sync.Once
+	dataDir          string
+	credentialsDir   string
+	opaServerRunning bool
+	once             sync.Once
 }
 
 func NewCollector(dataDir string, credentialsDir string) *Collector {
 	return &Collector{
-		dataDir:        dataDir,
-		credentialsDir: credentialsDir,
+		dataDir:          dataDir,
+		credentialsDir:   credentialsDir,
+		opaServerRunning: false,
```

Comment:
I just wonder, if we could get remove this variable and change `updateOpaServerStatus` to `getOpaServerStatus: boolean`, would that work?